Microsoft/WebTemplateStudio
# Template tests part 2

How it works
- this script automates running yarn install, yarn start and yarn test for all the generated projects from generate-test.js and throwing any errors that may appear 
- Running yarn start on the projects can allow us to detect errors in server side code (thrown when we detect an error that has a traceback)
- Running yarn test on the projects allows us to also the available unit tests for the frontend frameworks (currently only angular has a working unit test)
- After the test finishes (and if an error isn't thrown), all the generated projects from this test will be deleted  

How to test:
**1st iteration** -> this test should pass 
- cd WebTemplateStudio/src/extension 
- run yarn template-tests to generate the templates and run the test

**2nd iteration** -> this test should fail
- go to any angular project and navigate to app.component.ts 
- change the title attribute inside class AppComponent to a different string 
- run yarn template-tests 
- script should throw an error after it finishes running yarn test on that project and terminate

What to expect:
- takes about 10 minutes in total for both generate-test and run-tests to finish running
